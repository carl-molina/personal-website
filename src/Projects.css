a.Projects-project {
  text-decoration: none;
  transition: background-color 0.33s ease;
  color: inherit;
}

a.Projects-project:visited {
  text-decoration: none;
  transition: background-color 0.33s ease;
  color: inherit;
}

h2.Project-title {
  transition: background-color 0.33s ease;
}

h2.Project-title:hover {
  background-color: rgb(61, 129, 112);
}

.Projects-gif {
  width: 100%;
}

