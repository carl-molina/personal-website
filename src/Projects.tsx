import React from "react";
import "./Projects.css";
import joblyGif from "../public/joblyGif.gif";
import pixlyGif from "../public/pixlyGif.gif";
import plantGif from "../public/plantAppGif.gif";

/** Projects: presentational component for projects.
 *
 * Props: none
 * State: none
 *
 * App -> RoutesList -> Projects
 */

function Projects(): JSX.Element {

    return (
        <div>
            <div className="Projects-container">
                <a
                    href="https://jobly.carlmolina.com/"
                    target="_blank"
                    rel="noopener"
                    className="Projects-project">
                    <h2 className="Project-title">
                        Jobly üë©üèΩ‚Äçüíº
                    </h2>
                </a>
                <a href="https://carlmolina-jobly.onrender.com/"
                    target="_blank"
                    rel="noopener"
                >
                    <img src={joblyGif} className="Projects-gif" />
                </a>
                <p>Job board where users can sign up, browse companies,
                    and apply to jobs. Companies can post new jobs. Protects
                    routes on unauthorized users. Manages React state. Backend
                    authentication using jsonschema and JWT. Additional tech:
                    React, React Router, Fetch API, Node.js, Express,
                    PostgreSQL, ElephantSQL, Render.&nbsp;
                    <a
                        href="https://github.com/carl-molina/jobly"
                        target="_blank"
                        rel="noopener"
                    >
                        Github repo.
                    </a>
                </p>
            </div>
            <div className="Projects-container">
                <a href="https://pixly.carlmolina.com/"
                    target="_blank"
                    rel="noopener"
                    className="Projects-project">
                    <h2 className="Project-title">
                        Pix.ly üì∑
                    </h2>
                </a>
                <a href="https://carl-molina-pixly.onrender.com/"
                    target="_blank"
                    rel="noopener"
                >
                    <img src={pixlyGif} className="Projects-gif" />
                </a>
                <p>Image site that lets users upload images, view other images,
                    search EXIF data, and view a photo's EXIF data.
                    Uses AWS S3 to store and retrieve image files. Additional
                    tech: Python, Flask, Jinja, SQLAlchemy, JS,
                    Pillow.
                    {/* TODO: add Github repo */}
                </p>
            </div>
            <div className="Projects-container">
                <a href="https://plantapp.carlmolina.com/"
                    target="_blank"
                    rel="noopener"
                    className="Projects-project">
                    <h2 className="Project-title">
                        Plant App üå±
                    </h2>
                </a>
                <a href="https://carl-molina-plant-app.onrender.com/"
                    target="_blank"
                    rel="noopener"
                >
                    <img src={plantGif} className="Projects-gif" />
                </a>
                <p>Plant search site that lets users search plants, view care
                    details, and bookmark individual plants. Additional tech:
                    Perenual API, Flask, Jinja, SQLAlchemy, JS, jQuery, Axios,
                    WTForm, Flask Session, Bcrypt, PostgreSQL.
                    {/* TODO: add Github repo */}
                </p>
            </div>
        </div>
    );
}

export default Projects;
