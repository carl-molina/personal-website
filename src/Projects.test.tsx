import React from "react";

import { test, afterEach, describe, expect, vi } from "vitest";
import { render, fireEvent, screen } from '@testing-library/react';
import Projects from "./Projects";


describe("Projects smoke test", function () {
    test('renders without crashing', function () {
        render(<Projects />);
    });
});


describe("Projects snapshot test", function () {
    test('matches snapshot', function () {
        const { container } = render(<Projects />);
        expect(container).toMatchSnapshot();
    });
});


// describe("Projects renders correct data", function () {
//     test('contains expected text', function () {
//         const res = render(<Projects />);
//         expect(res.queryByText('Welcome back CARL!')).toBeInTheDocument();
//         expect(res.queryByText('❌ Include a number')).toBeInTheDocument();
//         expect(res.queryByText('❌ Password must match')).toBeInTheDocument();
//     });

//     test('button attribute set to disabled on initial render', function () {
//         const { container } = render(<Projects />);
//         const btn = container.querySelector('button')!;
//         expect(btn.getAttribute('disabled')).toBeDefined();
//         expect(btn).toBeDisabled();
//     });
// });


// describe("Projects valid inputting and submitting", function () {
//     test('returns valid form changes andcalled',
//         async function () {
//             const { container } = render(
//                 <Projects />
//             );

//             const submitBtn = container.querySelector("button")!;
//             const newPwInput = container.querySelector(
//                 `input[name="newPw"]`) as HTMLInputElement;
//             const confirmPwInput = container.querySelector(
//                 `input[name="confirmNewPw"]`) as HTMLInputElement;

//             expect(submitBtn).toContainHTML("Reset Password");
//             expect(newPwInput).toHaveValue("");

//             fireEvent.change(newPwInput, { target: { value: "ABC123abc" } });

//             expect(newPwInput).toHaveValue("ABC123abc");
//             expect(confirmPwInput).toHaveValue("");

//             fireEvent.change(
//                 confirmPwInput,
//                 {
//                     target: {
//                         value: "ABC123abc"
//                     }
//                 }
//             );

//             expect(confirmPwInput).toHaveValue("ABC123abc");

//             fireEvent.click(submitBtn);

//             expect.toHaveBeenCalledTimes(1);
//         });

//     test('returns valid form changes and form submit',
//         async function () {
//             const { container } = render(
//                 <Projects{checkSubmit} />
//             );

//             function checkSubmit(formData: IPasswordFormData): void {
//                 expect(formData.newPw).toEqual("ABC123abc");
//                 expect(formData.confirmNewPw).toEqual("ABC123abc");
//             }

//             const submitBtn = container.querySelector("button")!;
//             const newPwInput = container.querySelector(
//                 `input[name="newPw"]`) as HTMLInputElement;

//             expect(newPwInput).toHaveValue("");

//             fireEvent.change(newPwInput, { target: { value: "ABC123abc" } });
//             const confirmPw = container.querySelector(
//                 `input[name="confirmNewPw"]`) as HTMLInputElement;

//             expect(confirmPw).toHaveValue("");

//             fireEvent.change(confirmPw, { target: { value: "ABC123abc" } });
//             fireEvent.click(submitBtn);
//         });
// });


// describe("Projects invalid inputs", function () {
//     test("input does not meet 1 uppercase letter password requirement",
//         async function () {
//             render(<Projects />);

//             const submitBtn = screen.getByRole(
//                 'button',
//                 {
//                     name: 'Reset Password'
//                 }
//             );

//             const newPwInput = screen.getByRole(
//                 'textbox',
//                 {
//                     name: 'Enter new password *'
//                 }
//             );

//             expect(submitBtn).toBeDisabled();
//             expect(newPwInput).toHaveValue("");

//             fireEvent.change(newPwInput, { target: { value: "abc123test" } });
//             await screen.findByText('❌ Include an uppercase letter');
//             expect(submitBtn).toBeDisabled();
//         });
//     test("input does not meet 1 lowercase letter password requirement",
//         async function () {
//             render(<Projects />);

//             const submitBtn = screen.getByRole(
//                 'button',
//                 {
//                     name: 'Reset Password'
//                 }
//             );

//             const newPwInput = screen.getByRole(
//                 'textbox',
//                 {
//                     name: 'Enter new password *'
//                 }
//             );

//             expect(submitBtn).toBeDisabled();
//             expect(newPwInput).toHaveValue("");

//             fireEvent.change(newPwInput, { target: { value: "ABC123TEST" } });
//             await screen.findByText('❌ Include a lowercase letter');
//             expect(submitBtn).toBeDisabled();
//         });
//     test("input does not meet 1 number password requirement",
//         async function () {
//             render(<Projects />);

//             const submitBtn = screen.getByRole(
//                 'button',
//                 {
//                     name: 'Reset Password'
//                 }
//             );

//             const newPwInput = screen.getByRole(
//                 'textbox',
//                 {
//                     name: 'Enter new password *'
//                 }
//             );

//             expect(submitBtn).toBeDisabled();
//             expect(newPwInput).toHaveValue("");

//             fireEvent.change(newPwInput, { target: { value: "ABCdefGHI" } });
//             await screen.findByText('❌ Include a number');
//             expect(submitBtn).toBeDisabled();
//         });
//     test("newPw does not match confirmNewPw", async function () {
//         render(<Projects />);

//         const submitBtn = screen.getByRole(
//             'button',
//             {
//                 name: 'Reset Password'
//             }
//         );
//         const newPwInput = screen.getByRole(
//             'textbox',
//             {
//                 name: 'Enter new password *'
//             }
//         );
//         const confirmPwInput = screen.getByRole(
//             'textbox',
//             {
//                 name: 'Confirm password *'
//             }
//         );

//         expect(submitBtn).toBeDisabled();
//         expect(newPwInput).toHaveValue("");
//         expect(confirmPwInput).toHaveValue("");

//         fireEvent.change(newPwInput, { target: { value: "ABC123abc" } });
//         fireEvent.change(confirmPwInput, { target: { value: "ABC123test" } });
//         await screen.findByText('❌ Password must match');
//         expect(submitBtn).toBeDisabled();
//     });
// });
