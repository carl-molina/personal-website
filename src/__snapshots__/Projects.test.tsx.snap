// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Projects snapshot test > matches snapshot 1`] = `
<div>
  <div>
    <div
      class="Projects-container"
    >
      <a
        class="Projects-project"
        href="https://jobly.carlmolina.com/"
        rel="noopener"
        target="_blank"
      >
        <h2
          class="Project-title"
        >
          Jobly 👩🏽‍💼
        </h2>
      </a>
      <a
        href="https://carlmolina-jobly.onrender.com/"
        rel="noopener"
        target="_blank"
      >
        <img
          class="Projects-gif"
          src="/public/joblyGif.gif"
        />
      </a>
      <p>
        Job board where users can sign up, browse companies, and apply to jobs. Companies can post new jobs. Protects routes on unauthorized users. Manages React state. Backend authentication using jsonschema and JWT. Additional tech: React, React Router, Fetch API, Node.js, Express, PostgreSQL, ElephantSQL, Render. 
        <a
          href="https://github.com/carl-molina/jobly"
          rel="noopener"
          target="_blank"
        >
          GitHub.
        </a>
      </p>
    </div>
    <div
      class="Projects-container"
    >
      <a
        class="Projects-project"
        href="https://pixly.carlmolina.com/"
        rel="noopener"
        target="_blank"
      >
        <h2
          class="Project-title"
        >
          Pix.ly 📷
        </h2>
      </a>
      <a
        href="https://carl-molina-pixly.onrender.com/"
        rel="noopener"
        target="_blank"
      >
        <img
          class="Projects-gif"
          src="/public/pixlyGif.gif"
        />
      </a>
      <p>
        Image site that lets users upload images, view other images, search EXIF data, and view a photo's EXIF data. Uses AWS S3 to store and retrieve image files. Additional tech: Python, Flask, Jinja, SQLAlchemy, JS, Pillow. 
        <a
          href="https://github.com/carl-molina/pix.ly"
          rel="noopener"
          target="_blank"
        >
          GitHub.
        </a>
      </p>
    </div>
    <div
      class="Projects-container"
    >
      <a
        class="Projects-project"
        href="https://plantapp.carlmolina.com/"
        rel="noopener"
        target="_blank"
      >
        <h2
          class="Project-title"
        >
          Plant App 🌱
        </h2>
      </a>
      <a
        href="https://carl-molina-plant-app.onrender.com/"
        rel="noopener"
        target="_blank"
      >
        <img
          class="Projects-gif"
          src="/public/plantAppGif.gif"
        />
      </a>
      <p>
        Plant search site that lets users search plants, view care details, and bookmark individual plants. Additional tech: Perenual API, Flask, Jinja, SQLAlchemy, JS, jQuery, Axios, WTForm, Flask Session, Bcrypt, PostgreSQL. 
        <a
          href="https://github.com/carl-molina/plant-app"
          rel="noopener"
          target="_blank"
        >
          GitHub.
        </a>
      </p>
    </div>
  </div>
</div>
`;
